pkg/river/encoding/encoding.go

```diff
@@ -0,0 +1,139 @@
+package encoding
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/grafana/agent/pkg/river/token/builder"
+
+	"github.com/grafana/agent/pkg/river/internal/rivertags"
+	"github.com/grafana/agent/pkg/river/internal/value"
+)
+
+const attr = "attr"
+const object = "object"
+
+// ConvertRiverBlock is used to convertBase arguments, exports, health and debuginfo.
+func ConvertRiverBlock(input interface{}) ([]interface{}, error) {
```

Comment:
nit: `input` will always be a River body, never a River block. 

River blocks are named bodies, and the name information is always lost from a Go value. This is why in `river/token/builder` you have to [create a block separately](https://github.com/grafana/agent/blob/9e279c9bb43291f4d1b8242334219519779562eb/pkg/river/token/builder/builder.go#L271-L280) before you can [encode the body into it](https://github.com/grafana/agent/blob/9e279c9bb43291f4d1b8242334219519779562eb/pkg/river/token/builder/builder.go#L119-L132).

Reply:
bump